lib/net/ssh/command_stream.rb

```diff
@@ -74,31 +77,41 @@ def initialize(ssh, cmd = nil, pty: false, cleanup: false)
       end
 
       channel.on_open_failed do |ch, code, desc|
-        raise Net::SSH::ChannelOpenFailed.new(code, 'Session channel open failed')
+        error = Net::SSH::ChannelOpenFailed.new(code, 'Session channel open failed')
+        handle_error(error: error)
       end
 
       self.monitor = Thread.new do
-        while(true)
-          next if not self.rsock.has_read_data?(1.0)
-          buff = self.rsock.read(16384)
-          break if not buff
-          verify_channel
-          self.channel.send_data(buff) if buff
+        begin
+          Kernel.loop do
+            next if not self.rsock.has_read_data?(1.0)
+
+            buff = self.rsock.read(16384)
+            break if not buff
+
+            verify_channel
+            self.channel.send_data(buff) if buff
+          end
+        rescue ::StandardError => e
+          handle_error(error: e)
         end
       end
 
-      while true
-        rssh.process(0.5) { true }
+      begin
+        Kernel.loop { rssh.process(0.5) { true } }
+      rescue ::StandardError => e
+        handle_error(error: e)
       end
 
       # Shut down the SSH session if requested
       if !rcmd.nil? && rcleanup
         rssh.close
       end
     end
-  rescue ::Exception => e
+    self.thread.abort_on_exception = true
```

Comment:
I'll check if this is still necessary with the current implementation of `handle_error`

Reply:
I executed the module vs. a broken Windows 10 OpenSSH Server and Ubuntu, and both virtual machines worked as expected _without_ this line, so I am reverting it 👍